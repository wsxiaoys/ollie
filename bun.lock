{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "ollie",
      "dependencies": {
        "@commander-js/extra-typings": "^14.0.0",
        "commander": "^14.0.1",
      },
      "devDependencies": {
        "@types/bun": "latest",
        "ollie": ".",
      },
      "peerDependencies": {
        "typescript": "^5",
      },
    },
  },
  "packages": {
    "@commander-js/extra-typings": ["@commander-js/extra-typings@14.0.0", "", { "peerDependencies": { "commander": "~14.0.0" } }, "sha512-hIn0ncNaJRLkZrxBIp5AsW/eXEHNKYQBh0aPdoUqNgD+Io3NIykQqpKFyKcuasZhicGaEZJX/JBSIkZ4e5x8Dg=="],

    "@types/bun": ["@types/bun@1.2.23", "", { "dependencies": { "bun-types": "1.2.23" } }, "sha512-le8ueOY5b6VKYf19xT3McVbXqLqmxzPXHsQT/q9JHgikJ2X22wyTW3g3ohz2ZMnp7dod6aduIiq8A14Xyimm0A=="],

    "@types/node": ["@types/node@24.6.0", "", { "dependencies": { "undici-types": "~7.13.0" } }, "sha512-F1CBxgqwOMc4GKJ7eY22hWhBVQuMYTtqI8L0FcszYcpYX0fzfDGpez22Xau8Mgm7O9fI+zA/TYIdq3tGWfweBA=="],

    "@types/react": ["@types/react@19.1.15", "", { "dependencies": { "csstype": "^3.0.2" } }, "sha512-+kLxJpaJzXybyDyFXYADyP1cznTO8HSuBpenGlnKOAkH4hyNINiywvXS/tGJhsrGGP/gM185RA3xpjY0Yg4erA=="],

    "bun-types": ["bun-types@1.2.23", "", { "dependencies": { "@types/node": "*" }, "peerDependencies": { "@types/react": "^19" } }, "sha512-R9f0hKAZXgFU3mlrA0YpE/fiDvwV0FT9rORApt2aQVWSuJDzZOyB5QLc0N/4HF57CS8IXJ6+L5E4W1bW6NS2Aw=="],

    "commander": ["commander@14.0.1", "", {}, "sha512-2JkV3gUZUVrbNA+1sjBOYLsMZ5cEEl8GTFP2a4AVz5hvasAMCQ1D2l2le/cX+pV4N6ZU17zjUahLpIXRrnWL8A=="],

    "csstype": ["csstype@3.1.3", "", {}, "sha512-M1uQkMl8rQK/szD0LNhtqxIPLpimGm8sOBwU7lLnCpSbTyY3yeU1Vc7l4KT5zT4s/yOxHH5O7tIuuLOCnLADRw=="],

    "ollie": ["ollie@file:", { "dependencies": { "@commander-js/extra-typings": "^14.0.0", "commander": "^14.0.1" }, "devDependencies": { "@types/bun": "latest", "ollie": "." }, "peerDependencies": { "typescript": "^5" }, "bin": { "ollie": "src/cli.ts" } }],

    "typescript": ["typescript@5.9.2", "", { "bin": { "tsc": "bin/tsc", "tsserver": "bin/tsserver" } }, "sha512-CWBzXQrc/qOkhidw1OzBTQuYRbfyxDXJMVJ1XNwUHGROVmuaeiEm3OslpZ1RV96d7SKKjZKrSJu3+t/xlw3R9A=="],

    "undici-types": ["undici-types@7.13.0", "", {}, "sha512-Ov2Rr9Sx+fRgagJ5AX0qvItZG/JKKoBRAVITs1zk7IqZGTJUwgUr7qoYBpWwakpWilTZFM98rG/AFRocu10iIQ=="],

    "ollie/ollie": ["ollie@file:.", {}],
  }
}
