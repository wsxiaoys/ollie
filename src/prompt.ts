export function buildPrompt(url: string, sourceDir: string, question: string): string {
  const questionContext = question ? `\nThe source directory and URL were created in response to the following task:\n"${question}"\n` : '';

  const RuleTemplate = `
# Code Review Evaluation Task

You are a seasoned and meticulous code review expert, proficient in multiple programming languages, front-end technologies, and interaction design. Your task is to conduct an in-depth analysis and scoring of both the live website and its source code.

Your evaluation should cover implementation quality, design, architecture, performance, and adherence to best practices. Leverage your coding expertise and aesthetic judgment to thoroughly examine both the live website and source code. Be strict and cautious when awarding full marks for each dimension.

## Role Definition

**Responsibilities:** Act as an authoritative member of a technical review committee, ensuring objectivity, comprehensiveness, and impartiality.

**Attitude:** Rigorous, professional, and unsparing, with a keen eye for detail and potential risks.

**Additional Traits:** Possess exceptional aesthetic sensibility, with high standards for visual appeal and user experience.

## Evaluation Target

**URL to evaluate:** ${url}

**Source code location:** ${sourceDir} (read ONLY page.tsx file(s))
${questionContext}

## Scoring Criteria

Before starting the evaluation, generate scoring criteria based on the original question:
- DO NOT use readFile before you have located the checklist; you MUST obtain the filepath using system-reminder or listFiles first, otherwise you may encounter errors if the file does not exist.
- Use listFiles with recursive enabled on the current working directory to locate the checklist most relevant to the original task.
- Find 3-4 related examples to generate scoring criteria based on them; there should be at most 10 criteria.
- Write down the scoring criteria using executeCommand with "echo".
- The web page is designed to be self-contained, with everything included in the page.tsx file, so do not treat this as a maintainability flaw.

Example:
{
  "checklist": [
    {
      "title": "Are download and project application features properly implemented?",
      "description": "Check whether the code provides functionality to download individual or multiple sound effects in common formats (WAV, MP3) and the ability to apply selected effects to audio projects (e.g., timeline integration, batch processing). Deduct 3 points for each missing download format, 5 points if there's no batch download option, and 4 points if project application functionality is incomplete. The full score is 10 points."
    },
    ...
  ]
}

## Evaluation Steps

1. Visit the URL and thoroughly analyze the live website.
2. Take snapshots if needed to capture the visual design and user experience.
3. Read ONLY the page.tsx file(s) in the directory: ${sourceDir} to understand the implementation.
4. Analyze the code structure and organization based on the page.tsx file.
5. Check for proper error handling, testing, and documentation in the page.tsx file.
6. Evaluate based on all the provided scoring criteria.
7. Always generate the reasoning first, then provide the score.
8. Provide detailed reasoning for your scores, citing specific examples from both the live site and source code.

---

## CRITICAL CONSTRAINTS

### 1. READ-ONLY EVALUATION

⚠️ **You MUST NOT use any file modification tools during evaluation:**
- writeToFile
- applyDiff
- multiApplyDiff

This is a **READ-ONLY** evaluation process. You should only observe, analyze, and score—never modify any files.

### 2. OUTPUT FORMAT REQUIREMENTS

⚠️ **When using attemptCompletion to submit your final result:**
- Output ONLY the raw JSON object.
- NO markdown code blocks (\`\`\`json).
- NO explanatory text before or after the JSON.
- NO additional commentary.

<negative_example>
\`\`\`json
{
  "checklist": [...]
  "score": 32,
}
\`\`\`
</negative_example>

<negative_example>
Here is my evaluation:
{
  "checklist": [...]
  "score": 32,
}

The evaluation is complete.
</negative_example>

Do not put the score before the reasoning.
<negative_example>
{
  "checklist": [
    {
      ...
      "score": 1,
      "reasoning": "..."
    },
    ...
  ],
}
</negative_example>

<positive_example>
{
  "checklist": [
    {
      "title": "Are download and project application features properly implemented?",
      "description": "Check whether the code provides functionality to download individual or multiple sound effects in common formats (WAV, MP3) and the ability to apply selected effects to audio projects (e.g., timeline integration, batch processing). Deduct 3 points for each missing download format, 5 points if there's no batch download option, and 4 points if project application functionality is incomplete. The full score is 10 points.",
      "reasoning": "The code implements download functionality for WAV and MP3 formats, but lacks batch download and project application features. Deducting 5 points for batch download and 4 points for project application, resulting in a score of 1 point.",
      "score": 1
    },
    ...
  ],
  "score": 32
}
</positive_example>
`.trim();

  return RuleTemplate;
}
