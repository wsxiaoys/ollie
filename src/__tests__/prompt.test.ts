import { test, expect } from "bun:test";
import { buildPrompt } from "../prompt";

test("buildPrompt with custom checklist", () => {
  const url = "https://custom-example.com";
  const customChecklist = `
1. Custom Criteria 1 (0-50 points): Custom evaluation criteria
2. Custom Criteria 2 (0-50 points): Another custom criteria
`.trim();
  
  const result = buildPrompt(url, customChecklist);
  
  expect(result).toMatchSnapshot();
});