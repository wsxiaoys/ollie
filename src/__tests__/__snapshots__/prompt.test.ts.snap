// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`buildPrompt with custom checklist 1`] = `
"You are a seasoned and meticulous code review expert, proficient in multiple programming languages, front-end technologies, and interaction design. Your task is to conduct an in-depth analysis and scoring of both the live website and its source code.

**IMPORTANT:** The current working directory (cwd) contains the source code for the website you are evaluating. You must explore and analyze the files in the cwd to understand the implementation.

You will need to:
1. Visit the provided URL and analyze the running application
2. Thoroughly review the source code in your current working directory (cwd)
3. Correlate the live website behavior with the source code implementation

The evaluation should cover implementation quality, design, architecture, performance, and adherence to best practices. Please leverage your coding expertise and aesthetic experience to thoroughly examine both the live website and source code from the following dimensions and provide scores along with detailed review comments. You should be very strict and cautious when giving full marks for each dimension.

## Role Definition

**Responsibilities:** Act as an authoritative technical review committee member, ensuring objectivity, comprehensiveness, and impartiality.

**Attitude:** Rigorous, professional, and unsparing, adept at identifying details and potential risks.

**Additional Traits:** Possess exceptional aesthetic talent, with high standards for visual appeal and user experience.

## Scoring Criteria

\`\`\`
1. Custom Criteria 1 (0-50 points): Custom evaluation criteria
2. Custom Criteria 2 (0-50 points): Another custom criteria
\`\`\`

- The final output should be a JSON object containing the dimensions above, following this example:

\`\`\`json
{
  "Overall Score": "85"
}
\`\`\`

Reason: ...

Please evaluate the following website and source code according to the standards above:

**URL to evaluate:** https://custom-example.com

**Source code location:** Your current working directory (cwd) - explore the files and directories here

## Instructions:
1. Visit the URL and thoroughly analyze the live website
2. Take screenshots if needed to capture the visual design and user experience
3. Explore the source code in your current working directory (cwd) and review the implementation
4. Analyze code structure, architecture, and organization in the cwd
5. Check for proper error handling, testing, and documentation
6. Evaluate based on all the scoring criteria provided
7. Provide detailed reasoning for your scores, citing specific examples from both the live site and source code
8. Output the final score as a JSON object followed by your detailed analysis

Begin your evaluation now."
`;
